src/microbe_stage/MembraneWaterRipple.cs

```diff
@@ -0,0 +1,452 @@
+    public float Phase = 0.2f;
+
+    /*Controls how quickly ripples dissipate
+    higher values = less viscous appearance*/
+
+    [Export]
+    public float Attenuation = 0.998f;
+
+    [Export]
+    public string ShaderPath = "res://shaders/MembraneWaterRipple.gdshader";
+
+    [Export]
+    public bool DynamicScaling = true;
+
+    [Export]
+    public float ScaleCheckInterval = 0.5f;
+
+    [Export]
+    public float RadiusMultiplier = 2.5f;
+
+    [Export]
+    public int MaxMeshSubdivision = 100; // Reduced for better performane can be increased for a more detailed effect
+
+    [Export]
+    public float VerticalOffset = -0.15f; // How far below the membrane the water plane should be
+
+    // Default radius if we can't determine it from the membrane
+    private const float DEFAULT_MEMBRANE_RADIUS = 5.0f;
+    private MeshInstance3D? waterPlane;
+    private ShaderMaterial? waterMaterial;
+    private Membrane? parentMembrane;
+
+    // Trail tracking variables
+    private Vector3 lastPosition;
+    private Vector3 previousPosition;
+    private float lastMovementTime;
+    private Vector2 currentDirection = Vector2.Right;
+    private float currentSpeed;
+
+    // State tracking
+    private float lastScaleCheckTime;
+    private float lastCheckedMembraneRadius;
+    private bool isSetup;
+
+    // Called when the node enters the scene tree for the first time
+    public override void _Ready()
+    {
+        // Initialize the effect
+        SetProcess(true);
+        SetPhysicsProcess(true);
+
+        // Find the parent membrane
+        parentMembrane = GetParent<Membrane>();
+        if (parentMembrane == null)
+        {
+            GD.PrintErr("MembraneWaterRipple must be a child of a Membrane");
+            QueueFree();
+            return;
+        }
+
+        // Initialize position tracking for movement calculation
+        lastPosition = parentMembrane.GlobalPosition;
+        previousPosition = lastPosition;
+        lastMovementTime = Time.GetTicksMsec() / 1000.0f;
+        CreateWaterPlane();
+
+        // Set initial position
+        UpdatePosition();
+    }
+
+    // Called every frame for visual updates
+    public override void _Process(double delta)
+    {
+        if (!EnableEffect || !isSetup || parentMembrane == null)
+            return;
+
+        try
+        {
+            // Update position every frame to ensure perfect alignment
+            UpdatePosition();
+
+            // Update shader time with exact original timing
+            if (waterMaterial != null)
+            {
+                float currentTime = Time.GetTicksMsec() / 1000.0f;
+                waterMaterial.SetShaderParameter("time_offset", currentTime);
+            }
+        }
+        catch (Exception e)
+        {
+            GD.PrintErr("Error in water ripple process: " + e.Message);
+        }
+    }
+
+    // Called on physics process for movement and simulation updates
+    public override void _PhysicsProcess(double delta)
```

Comment:
Is there an actual performance improvement from splitting the processing into two parts? I would guess that as you already check the time elapsed here, it would be more performance friendly to put all of the content from here into _Process and using a double variable to keep track when the extra checks need to run.